service/resource/certificate/certificate_test.go

```diff
@@ -1 +1,232 @@
 package certificate
+
+import (
+	"testing"
+	"time"
+
+	"github.com/spf13/afero"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+
+	"github.com/giantswarm/prometheus-config-controller/service/prometheus"
+)
+
+// Test_Resource_Certificate_New tests the New function.
+func Test_Resource_Certificate_New(t *testing.T) {
+	prometheusReloaderConfig := prometheus.DefaultConfig()
+
+	prometheusReloaderConfig.K8sClient = fake.NewSimpleClientset()
+	prometheusReloaderConfig.Logger = microloggertest.New()
+
+	prometheusReloaderConfig.Address = "http://127.0.0.1:9090"
+	prometheusReloaderConfig.ConfigMapKey = "prometheus.yml"
+	prometheusReloaderConfig.ConfigMapName = "prometheus"
+	prometheusReloaderConfig.ConfigMapNamespace = "monitoring"
+	prometheusReloaderConfig.MinimumReloadTime = 2 * time.Minute
+
+	prometheusReloader, _ := prometheus.New(prometheusReloaderConfig)
+
+	tests := []struct {
+		config func() Config
```

Comment:
I don't think it has to be func. DefaultConfig returns a single value so you can use it in struct initialization.

Reply:
This is true! I copy and pasted this from elsewhere, where I needed to do actual work in the function. I'm lazy :(